src/baseindicator.js

```diff
@@ -64,12 +64,32 @@ var CPUFreqBaseIndicator = class CPUFreqBaseIndicator {
 
         this.lblActive = (this.settings.get_boolean('show-freq-in-taskbar'));
         this.lblUnit = (this.settings.get_boolean('taskbar-freq-unit-ghz'));
+        this.iconActive = (this.settings.get_boolean('show-icon-in-taskbar'));
+        this.arrowActive = (this.settings.get_boolean('show-arrow-in-taskbar'));
 
-        this.hbox.add_actor(icon);
-        this.hbox.add_actor(PopupMenu.arrowIcon(St.Side.BOTTOM));
+        this.hbox.add_actor(this.icon);
+        this.arrow = PopupMenu.arrowIcon(St.Side.BOTTOM);
+        this.hbox.add_actor(this.arrow);
```

Comment:
Should this code only add the icon, label, and arrow if they are active or even better, call createIndicator?

Reply:
Yes, good idea. This makes the code actually a lot smaller.